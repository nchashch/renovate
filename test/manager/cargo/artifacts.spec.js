const fs = require('fs');
const upath = require('upath');
const cargo = require('../../../lib/manager/cargo/artifacts');
const { extractPackageFile } = require('../../../lib/manager/cargo/extract');

const crateDir = 'test/_fixtures/cargo/example_crate';
const cargoTomlFilePath = upath.join(crateDir, 'Cargo.toml');
const cargoLockFilePath = upath.join(crateDir, 'Cargo.lock');
const cargoToml = fs.readFileSync(cargoTomlFilePath, 'utf8');
// const cargoLock = fs.readFileSync(cargoLockFilePath, 'utf8');
let config;

describe('cargo.getArtifacts()', () => {
  beforeEach(() => {
    jest.resetAllMocks();
    config = {
      localDir: crateDir,
    };
  });
  it('returns null by default', async () => {
    expect(await cargo.getArtifacts('Cargo.toml', [], '', config)).toBeNull();
  });
  it('returns null by default', async () => {
    // NOTE: In this case cargo.getArtifacts() returns null, because
    // updatedDeps array is empty
    // TODO: Figure out if this is correct behavior
    expect(await cargo.getArtifacts()).toBeNull();
  });
  it('catches errors', async () => {
    const dep = {
      depName: 'INVALID_DEP',
      currentValue: 'INVALID_DEP',
    };
    // NOTE: If config.localDir is not set getArtifacts throws an exception
    // TODO: Figure out if this is correct behavior, most likely not -- it should
    // probably default to current working directory
    expect(await cargo.getArtifacts('Cargo.toml', [dep], '', {})).toMatchSnapshot();
  });
  it('generates Cargo.lock', async () => {
    const updatedDeps = extractPackageFile(cargoToml, 'Cargo.toml').deps;
    expect(
      await cargo.getArtifacts('Cargo.toml', updatedDeps, cargoToml, config)
    ).toMatchSnapshot();
    fs.unlinkSync(cargoLockFilePath);
  });
});
