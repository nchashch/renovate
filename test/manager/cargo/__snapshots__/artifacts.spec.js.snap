// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cargo.getArtifacts() catches errors 1`] = `
Object {
  "lockFileError": Object {
    "lockFile": "Cargo.lock",
    "stderr": "The \\"directory\\" argument must be of type string. Received type undefined",
  },
}
`;

exports[`cargo.getArtifacts() generates Cargo.lock 1`] = `
Object {
  "file": Object {
    "contents": "[[package]]
name = \\"example_crate\\"
version = \\"0.1.0\\"
dependencies = [
 \\"libc 0.2.44 (registry+https://github.com/rust-lang/crates.io-index)\\",
]

[[package]]
name = \\"libc\\"
version = \\"0.2.44\\"
source = \\"registry+https://github.com/rust-lang/crates.io-index\\"

[metadata]
\\"checksum libc 0.2.44 (registry+https://github.com/rust-lang/crates.io-index)\\" = \\"10923947f84a519a45c8fefb7dd1b3e8c08747993381adee176d7a82b4195311\\"
",
    "name": "Cargo.lock",
  },
}
`;
